stripe/src/test/java/com/stripe/android/net/StripeApiHandlerTest.java

```diff
@@ -0,0 +1,124 @@
+package com.stripe.android.net;
+
+
+import android.os.Build;
+
+import com.stripe.android.Stripe;
+import com.stripe.android.exception.InvalidRequestException;
+import com.stripe.android.model.Card;
+import com.stripe.android.util.StripeNetworkUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * Test class for {@link StripeApiHandler}.
+ */
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = 23)
+public class StripeApiHandlerTest {
+
+    @Test
+    public void testGetApiUrl() {
+        String tokensApi = StripeApiHandler.getApiUrl();
+        assertEquals("https://api.stripe.com/v1/tokens", tokensApi);
+    }
+
+    @Test
+    public void getHeaders_withAllRequestOptions_properlyMapsRequestOptions() {
+        String fakePublicKey = "fake_public_key";
+        String idempotencyKey = "idempotency_rules";
+        String apiVersion = "2011-11-11";
+        RequestOptions requestOptions = RequestOptions.builder(fakePublicKey)
+                .setIdempotencyKey(idempotencyKey)
+                .setApiVersion(apiVersion)
+                .build();
+        Map<String, String> headerMap = StripeApiHandler.getHeaders(requestOptions);
+
+        assertNotNull(headerMap);
+        assertEquals("Bearer " + fakePublicKey, headerMap.get("Authorization"));
+        assertEquals(idempotencyKey, headerMap.get("Idempotency-Key"));
+        assertEquals(apiVersion, headerMap.get("Stripe-Version"));
+    }
+
+    @Test
+    public void getHeaders_withOnlyRequiredOptions_doesNotAddEmptyOptions() {
+        RequestOptions requestOptions = RequestOptions.builder("some_key").build();
+        Map<String, String> headerMap = StripeApiHandler.getHeaders(requestOptions);
+
+        assertNotNull(headerMap);
+        assertFalse(headerMap.containsKey("Idempotency-Key"));
+        assertTrue(headerMap.containsKey("Authorization"));
+        // Note: it contains this because RequestOptions always returns a value for api version
+        assertTrue(headerMap.containsKey("Stripe-Version"));
```

Comment:
We still don't have an official standard unfortunately.

I personally think that non-dynamic languages should tend to pass an actual version in because even if new things come back, they won't be able to handle them anyway. This is what stripe-go does already and what stripe-java probably should do.